### PR TITLE
fixes s3 stream not adding connection back to pool

### PR DESCRIPTION
**Pull request by Paul-Hess**
PDXDDT-838

### CODE REVIEW

**Comment by lkemmerer** [comment]
I think I was seeing this warning when the hub received timeouts while it was processing stuff on the bulk endpoint, too....somewhere in `ZipInputStream`, I think, we're not using a try-with-resources where we might need to. That's probably a different ticket, though...

File: src/main/java/com/flightstats/hub/dao/aws/S3AccessMonitor.java [hunk 1]
```diff
@@ -89,6 +89,7 @@ private String key() {
             return CompletableFuture.supplyAsync(() -> {
                 try (S3Object s3Object = hubS3Client
                         .getObject(new GetObjectRequest(bucketName, key()))) {
+                    s3Object.getObjectContent().read(new byte[s3Object.getObjectContent().available()]);
```
Comment (lkemmerer) [inline src/main/java/com/flightstats/hub/dao/aws/S3AccessMonitor.java]: would calling `s3Object.getObjectContent().close()` do roughly the same thing?

Comment (Paul-Hess) [reply src/main/java/com/flightstats/hub/dao/aws/S3AccessMonitor.java]: that would throw, if we explicitly close the connection without reading all the bytes out it blows up.

**Comment by Paul-Hess** [comment]
> I think I was seeing this warning when the hub received timeouts while it was processing stuff on the bulk endpoint, too....somewhere in `ZipInputStream`, I think, we're not using a try-with-resources where we might need to. That's probably a different ticket, though...

I remembered that I had come across those as I was searching for the culprit. I added a couple simple fixes for those to this as well

File: src/main/java/com/flightstats/hub/dao/aws/S3AccessMonitor.java (hunk 1 above)
Comment (lucasmo) [reply src/main/java/com/flightstats/hub/dao/aws/S3AccessMonitor.java]: This makes me leery given the unknown size of the bucket. Is the goal here to read the entire stream and discard it, much like an http request? In this case, available() is what's currently in the buffer, not the actual length of the document.

I'd do something like have a 4k buffer and read into it repeatedly until EOF.

File: src/main/java/com/flightstats/hub/dao/aws/S3BatchContentDao.java [hunk 2]
```diff
@@ -152,13 +153,15 @@ private Content getContent(ContentKey key, ZipInputStream zipStream, ZipEntry ne
...
+    @SneakyThrows
     private ZipInputStream getZipInputStream(String channel, ContentPathKeys minutePath) {
         ActiveTraces.getLocal().add("S3BatchContentDao.getZipInputStream");
         long start = System.currentTimeMillis();
         try {
             GetObjectRequest request = new GetObjectRequest(bucketName, getS3BatchItemsKey(channel, minutePath));
-            S3Object object = s3Client.getObject(request);
-            return new ZipInputStream(new BufferedInputStream(object.getObjectContent()));
+            try (S3Object object = s3Client.getObject(request)) {
+                return new ZipInputStream(new BufferedInputStream(object.getObjectContent()));
```
Comment (lucasmo) [inline src/main/java/com/flightstats/hub/dao/aws/S3BatchContentDao.java]: This is going to close the stream on exit.

https://github.com/aws/aws-sdk-java/blob/master/aws-java-sdk-s3/src/main/java/com/amazonaws/services/s3/model/S3Object.java#L223

File: src/main/java/com/flightstats/hub/dao/aws/S3AccessMonitor.java (hunk 1 above)
Comment (lucasmo) [reply src/main/java/com/flightstats/hub/dao/aws/S3AccessMonitor.java]: (If you have IOUtils available, you can readFully into a NullOutputStream to save on code, I believe)

File: src/main/java/com/flightstats/hub/dao/aws/S3LargeContentDao.java [hunk 3]
```diff
@@ -199,18 +199,20 @@ private Content getS3Object(String channelName, ContentKey key) throws IOExcepti
...
+            try (S3Object object = s3Client.getObject(request)) {
+                ObjectMetadata metadata = object.getObjectMetadata();
+                Map<String, String> userData = metadata.getUserMetadata();
+                Content.Builder builder = Content.builder();
+                String type = userData.get("type");
+                if (!type.equals("none")) {
+                    builder.withContentType(type);
+                }
+                builder.withContentKey(key);
+                builder.withStream(object.getObjectContent());
```
Comment (lucasmo) [inline src/main/java/com/flightstats/hub/dao/aws/S3LargeContentDao.java]: I think this is going to suffer from the same problem of closing the stream in the returned object.

File: src/main/java/com/flightstats/hub/dao/aws/S3BatchContentDao.java (hunk 2 above)
Comment (Paul-Hess) [reply src/main/java/com/flightstats/hub/dao/aws/S3BatchContentDao.java]: True, tests are broken.

**Pull request merged**